test/e2e/specs/editor/various/autosave.spec.js

```diff
@@ -0,0 +1,351 @@
+			.filter( { hasText: 'Draft saved' } )
+			.waitFor();
+		await page.keyboard.type( ' after save' );
+
+		// Trigger local autosave.
+		await page.evaluate( () =>
+			window.wp.data.dispatch( 'core/editor' ).autosave( { local: true } )
+		);
+
+		expect(
+			await page.evaluate( () => window.sessionStorage.length )
+		).toBeGreaterThanOrEqual( 1 );
+
+		// Intercept autosave request and abort it.
+		await page.route(
+			( url ) => url.href.includes( 'autosave' ),
+			async ( route ) => {
+				await route.abort( 'internetdisconnected' );
+			}
+		);
+		await page.evaluate( () =>
+			window.wp.data.dispatch( 'core/editor' ).autosave()
+		);
+
+		expect(
+			await page.evaluate( () => window.sessionStorage.length )
+		).toBe( 1 );
+	} );
+
+	test( 'should clear local autosave after successful save', async ( {
+		page,
+		pageUtils,
+	} ) => {
+		const notice = page
+			.getByRole( 'button', { name: 'Dismiss this notice' } )
+			.filter( { hasText: 'Draft saved' } );
+
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'before save' );
+		await pageUtils.pressKeys( 'primary+s' );
+		await notice.waitFor();
+		await page.keyboard.type( ' after save' );
+		await notice.click();
+
+		// Trigger local autosave.
+		await page.evaluate( () =>
+			window.wp.data.dispatch( 'core/editor' ).autosave( { local: true } )
+		);
+
+		expect(
+			await page.evaluate( () => window.sessionStorage.length )
+		).toBeGreaterThanOrEqual( 1 );
+
+		await pageUtils.pressKeys( 'primary+s' );
+		await notice.waitFor();
+
+		expect(
+			await page.evaluate( () => window.sessionStorage.length )
+		).toBe( 0 );
+	} );
+
+	test( "shouldn't clear local autosave if save fails", async ( {
+		page,
+		pageUtils,
+	} ) => {
+		const notice = page
+			.getByRole( 'button', { name: 'Dismiss this notice' } )
+			.filter( { hasText: 'Draft saved' } );
+
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'before save' );
+		await pageUtils.pressKeys( 'primary+s' );
+		await notice.waitFor();
+		await page.keyboard.type( ' after save' );
+		await notice.click();
+
+		// Trigger local autosave.
+		await page.evaluate( () =>
+			window.wp.data.dispatch( 'core/editor' ).autosave( { local: true } )
+		);
+
+		expect(
+			await page.evaluate( () => window.sessionStorage.length )
+		).toBeGreaterThanOrEqual( 1 );
+
+		await page.route(
+			( url ) =>
+				url.href.includes(
+					`rest_route=${ encodeURIComponent( '/wp/v2/posts/' ) }`
+				),
+			async ( route ) => {
+				await route.abort( 'internetdisconnected' );
+			}
+		);
+		await pageUtils.pressKeys( 'primary+s' );
+
+		await expect(
+			page.locator( '.components-notice__content' )
+		).toContainText( 'Updating failed. You are probably offline.' );
+		expect(
+			await page.evaluate( () => window.sessionStorage.length )
+		).toBe( 1 );
+	} );
+
+	// See https://github.com/WordPress/gutenberg/pull/17501.
+	test( "shouldn't conflict with server-side autosave", async ( {
```

Comment:
I wasn't able to run this test successfully. Either there's a bug in the core, or the test itself is flaky. Currently, it only passes because of a conditional early return.

Do you think we should skip it until the issue is resolved?